src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -467,6 +470,12 @@ def set_msal_telemetry(msal_telemetry):
         _session.msal_telemetry = msal_telemetry
 
 
+@decorators.suppress_all_exceptions()
+def set_user_agent(user_agent):
+    if user_agent:
+        _session.user_agent = user_agent
```

Comment:
We finally decide to allow overwrite the existing `user_agent`. Because if there're several http requests in one cli command execution, then most likely all the previous requests are for resource preparation. It makes more sense to record the user agent of last http request.